### PR TITLE
Fix multi-endpoint text decoder when on root mode

### PR DESCRIPTION
**Pull request by tkountis**
Fixes https://github.com/hazelcast/hazelcast/issues/14516

Fix Text decoder when on root mode to initialize the src buffer propertly.
Add memcached tests for multiendpoint config

### CODE REVIEW

**Comment by tkountis** [comment]
@olukas can you check this branch?

**Comment by vbekiaris** [comment]
run-lab-run

**Comment by tkountis** [comment]
Thanks @olukas - Yes, I saw the issue, I am pushing the changes shortly.

**Comment by tkountis** [comment]
run-lab-run

File: hazelcast/src/test/java/com/hazelcast/internal/ascii/MemcachedMultiendpointTest.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+import com.hazelcast.test.annotation.QuickTest;
+import org.junit.experimental.categories.Category;
+import org.junit.runner.RunWith;
+
+@RunWith(HazelcastParallelClassRunner.class)
+@Category(QuickTest.class)
+public class MemcachedMultiendpointTest
+        extends MemcachedTest {
+
+    public void setup() throws Exception {
```
Comment (vbekiaris) [inline hazelcast/src/test/java/com/hazelcast/internal/ascii/MemcachedMultiendpointTest.java]: minor: missing `@Override`?

**Comment by tkountis** [comment]
run-lab-run

**Comment by vbekiaris** [comment]
run-lab-run

**Comment by mmedenjak** [comment]
run-lab-run

**Comment by mmedenjak** [comment]
Merged as test failure is known issue unrelated to the change.

**Pull request merged**